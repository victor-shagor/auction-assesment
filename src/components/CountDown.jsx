import React from "react";
import { makeStyles } from "@mui/styles";
import Box from "@mui/material/Box";
import Button from "@mui/material/Button";
import ThumbsIcon from "@mui/icons-material/ThumbUpAltOutlined";
import image from "../assets/images/Image.png";
import logo from "../assets/images/Avatarphoto.png";

const useStyles = makeStyles({
  wrapper: {
    margin: "20px",
    borderRadius: "8px",
    padding: "20px",
    border: "1px solid #EAECF0",
    fontFamily: "IBMPlexRegular",
  },
  innerBox: {
    width: "100%",
    borderRadius: "8px",
    position: "relative",
  },
  img: {
    width: "100%",
    borderRadius: "8px",
  },
  logo: {
    position: "absolute",
    top: 150,
    left: 15,
    "@media(max-width:900px)": {
      display: "none",
    },
  },
  countDownWrapper: {
    marginLeft: "13rem",
    display: "flex",
    justifyContent: "space-between",
    alignItems: "center",
    "@media(max-width:900px)": {
      marginLeft: 0,
      display: "block",
    },
  },
  countdown: {
    fontFamily: "IBMPlexSemiBold",
    marginBottom: "5px",
  },
  flexBox: {
    display: "flex",
    alignItems: "center",
    marginBottom: "15px",
  },
  liveWrapper: {
    backgroundColor: "#FFFAEB",
    width: "80px",
    display: "flex",
    justifyContent: "space-around",
    alignItems: "center",
    height: "15px",
    padding: "5px",
    borderRadius: "5px",
    marginRight: "10px",
  },
  dotBox: {
    height: "6px",
    width: "6px",
    borderRadius: "50%",
    backgroundColor: "#F79009",
  },
});

const CountDown = () => {
  const classes = useStyles();
  return (
    <Box className={classes.wrapper}>
      <Box className={classes.innerBox}>
        <img src={image} alt="" className={classes.img} />
        <Box>
          <img src={logo} alt="" className={classes.logo} />
          <Box className={classes.countDownWrapper}>
            <Box>
              <p className={classes.countdown}>
                Starts in: 3 days : 2 hours : 24 minutes{" "}
              </p>
              <Box className={classes.flexBox}>
                <Box className={classes.liveWrapper}>
                  <Box className={classes.dotBox} />
                  <span> Not Live</span>
                </Box>
                <span>Layers Auction</span>
              </Box>
            </Box>
            <Button
              disableElevation
              size="small"
              variant="contained"
              sx={{
                textTransform: "none",
                color: "#344054",
                backgroundColor: "white",
                border: "1px solid #D0D5DD",
                borderRadius: "8px",
              }}
              startIcon={<ThumbsIcon />}
            >
              Accept Invite
            </Button>
          </Box>
        </Box>
      </Box>
    </Box>
  );
};

export default CountDown;
