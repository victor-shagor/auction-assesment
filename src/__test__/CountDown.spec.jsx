import { render, screen } from "@testing-library/react";
import CountDown from "../components/CountDown";

describe("countdown component", () => {
  it("should render countdown component", async () => {
    render(<CountDown />);

    expect(screen.getByTestId("countdown")).toBeInTheDocument();
  });

  it("renders correctly", () => {
    const { container } = render(
      <CountDown product={{ title: "test title" }} />
    );
    expect(container).toMatchSnapshot();
  });
});
