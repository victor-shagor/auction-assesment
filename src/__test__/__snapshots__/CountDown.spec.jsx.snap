// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`countdown component renders correctly 1`] = `
<div>
  <div
    class="makeStyles-wrapper-10 MuiBox-root css-0"
    data-testid="countdown"
  >
    <div
      class="makeStyles-innerBox-11 MuiBox-root css-0"
    >
      <img
        alt=""
        class="makeStyles-img-12"
        src="Image.png"
      />
      <div
        class="MuiBox-root css-0"
      >
        <img
          alt=""
          class="makeStyles-logo-13"
          src="Avatarphoto.png"
        />
        <div
          class="makeStyles-countDownWrapper-14 MuiBox-root css-0"
        >
          <div
            class="MuiBox-root css-0"
          >
            <p
              class="makeStyles-countdown-15"
            >
              Starts in: 3 days : 2 hours : 24 minutes
               
            </p>
            <div
              class="makeStyles-flexBox-16 MuiBox-root css-0"
            >
              <div
                class="makeStyles-liveWrapper-17 MuiBox-root css-0"
              >
                <div
                  class="makeStyles-dotBox-18 MuiBox-root css-0"
                />
                <span>
                   Not Live
                </span>
              </div>
              <span>
                Layers Auction
              </span>
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-disableElevation css-1y4a2p3-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeSmall css-y6rp3m-MuiButton-startIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ThumbUpAltOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M21 8h-6.31l.95-4.57.03-.32c0-.41-.17-.79-.44-1.06L14.17 1 7.59 7.59C7.22 7.95 7 8.45 7 9v10c0 1.1.9 2 2 2h9c.83 0 1.54-.5 1.84-1.22l3.02-7.05c.09-.23.14-.47.14-.73v-2c0-1.1-.9-2-2-2zm0 4-3 7H9V9l4.34-4.34L12.23 10H21v2zM1 9h4v12H1z"
                />
              </svg>
            </span>
            Accept Invite
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
